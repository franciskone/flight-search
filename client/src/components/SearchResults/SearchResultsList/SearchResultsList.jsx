import React from 'react';
import PropTypes from 'prop-types';
import withInfiniteScroll, { ArrayDataSource } from 'bpk-component-infinite-scroll';
import BpkButton from 'bpk-component-button';

import ItineraryItem, { ItineraryType } from '../ItineraryItem';
import { styleGetter } from '../../../util';

import STYLE from './SearchResultsList.scss';

const c = styleGetter(STYLE);

const SearchResultsList = ({ elements }) => (
  <div className={c('SearchResultsList')}>
    {
      elements.map(({
        legs, price, id, agent,
      }) => (
        <ItineraryItem
          key={id}
          id={id}
          agent={agent}
          legs={legs}
          price={price}
        />
      ))
    }
  </div>
);

SearchResultsList.propTypes = {
  elements: PropTypes.arrayOf(PropTypes.shape(ItineraryType)).isRequired,
};

const InfiniteList = withInfiniteScroll(SearchResultsList);

const SeeMoreButton = ({ onSeeMoreClick }) => (
  <div className={c('SeeMore')}>
    <BpkButton large onClick={onSeeMoreClick}>See More</BpkButton>
  </div>
);

export default ({ itineraries }) => {
  const dataSource = new ArrayDataSource(itineraries);
  
  return (
    <InfiniteList
      dataSource={dataSource}
      renderSeeMoreComponent={SeeMoreButton}
      seeMoreAfter={1}
    />
  );
};

