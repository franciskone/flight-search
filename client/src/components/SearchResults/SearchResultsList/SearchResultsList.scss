@import '~bpk-mixins';

.SearchResultsList {
  padding: $bpk-spacing-sm;
  max-width: 960px;
  margin: auto;
}
