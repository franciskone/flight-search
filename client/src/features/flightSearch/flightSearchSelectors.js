// TODO Franciskone: add tests
import { FLIGHT_SEARCH_REDUCER_NAME } from './flightSearchReducer';

const FALLBACK_VALUE = '---';
const flightSearchState = state => state[FLIGHT_SEARCH_REDUCER_NAME];

export const flightSearchIsLoadingSelector = state => flightSearchState(state).isLoading;
export const flightSearchQuerySelector = state => flightSearchState(state).query;
const flightSearchPlaceSelector = (state, placeId) => flightSearchState(state).placesById[placeId];

// TODO Franciskone: convert to reselectors
export const flightSearchQueryOriginCodeSelector = (state) => {
  const { OriginPlace } = flightSearchQuerySelector(state);
  
  if (OriginPlace) {
    const place = flightSearchPlaceSelector(state, OriginPlace);
    return place
      ? place.Code
      : FALLBACK_VALUE;
  }
  return FALLBACK_VALUE;
};

export const flightSearchQueryDestinationCodeSelector = (state) => {
  const { DestinationPlace } = flightSearchQuerySelector(state);
  
  if (DestinationPlace) {
    const place = flightSearchPlaceSelector(state, DestinationPlace);
    return place
      ? place.Code
      : FALLBACK_VALUE;
  }
  return FALLBACK_VALUE;
};

export const flightSearchQueryPassengersNumberSelector =
    state => flightSearchQuerySelector(state).Adults || 0;

export const flightSearchQueryPassengersCabinClassSelector =
    state => flightSearchQuerySelector(state).CabinClass || FALLBACK_VALUE;

export const flightSearchHasQuerySelector = state => !!flightSearchQuerySelector(state).SessionKey;
