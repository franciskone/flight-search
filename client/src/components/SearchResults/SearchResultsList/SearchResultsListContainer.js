import React from 'react';
import { connect } from 'react-redux';
import SearchResultsList from './SearchResultsList';
import { flightSearchItinerariesForResultsSelector } from '../../../features/flightSearch';


const mapStateToProps = state => ({
  itineraries: flightSearchItinerariesForResultsSelector(state),
});

const SearchResultsLisContainer = connect(mapStateToProps)(SearchResultsList);

export default SearchResultsLisContainer;
