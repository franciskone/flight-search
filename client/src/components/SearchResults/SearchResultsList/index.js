import SearchResultsListContainer from './SearchResultsListContainer';

export default SearchResultsListContainer;

