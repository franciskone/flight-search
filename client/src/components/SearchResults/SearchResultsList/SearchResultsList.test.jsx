import React from 'react';
import TestRenderer from 'react-test-renderer';

import { SearchResultsFullListPresentation } from './SearchResultsList';
import { flightSearchItinerariesSelectorOutput } from '../../../features/flightSearch/flightSearchSelectors.test';

describe('SearchResultsFullListPresentation', () => {
  it('should render correctly', () => {
    const tree = TestRenderer.create(
      <SearchResultsFullListPresentation
        elements={flightSearchItinerariesSelectorOutput}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
