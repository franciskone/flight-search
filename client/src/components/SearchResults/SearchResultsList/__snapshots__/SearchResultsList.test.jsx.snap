// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsFullListPresentation should render correctly 1`] = `
<div
  className="SearchResultsList"
>
  <div
    className="bpk-card bpk-card--padded ItineraryItem"
    role="button"
  >
    <div
      className="flight"
    >
      <div
        className="route"
      >
        <img
          alt="carrier logo"
          className="carrier-logo"
          src="https://logos.skyscnr.com/images/airlines/favicon/BA.png"
        />
        <div
          className="airport"
        >
          <span
            className="bpk-text bpk-text--lg airport__time"
          >
            19:35
          </span>
          <span
            className="bpk-text bpk-text--lg airport__name"
          >
            EDI
          </span>
        </div>
        <svg
          className="arrow"
          height="24"
          style={
            Object {
              "height": "1.5rem",
              "width": "1.5rem",
            }
          }
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
          />
        </svg>
        <div
          className="airport airport--destination"
        >
          <span
            className="bpk-text bpk-text--lg airport__time"
          >
            21:00
          </span>
          <span
            className="bpk-text bpk-text--lg airport__name"
          >
            LHR
          </span>
        </div>
      </div>
      <div
        className="details"
      >
        <span
          className="bpk-text bpk-text--base details__duration"
        >
          1h 25
        </span>
        <span
          className="bpk-text bpk-text--base details__stops details__stops--single"
        >
          Direct
        </span>
      </div>
    </div>
    <div
      className="flight"
    >
      <div
        className="route"
      >
        <img
          alt="carrier logo"
          className="carrier-logo"
          src="https://logos.skyscnr.com/images/airlines/favicon/U2.png"
        />
        <div
          className="airport"
        >
          <span
            className="bpk-text bpk-text--lg airport__time"
          >
            07:05
          </span>
          <span
            className="bpk-text bpk-text--lg airport__name"
          >
            LTN
          </span>
        </div>
        <svg
          className="arrow"
          height="24"
          style={
            Object {
              "height": "1.5rem",
              "width": "1.5rem",
            }
          }
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
          />
        </svg>
        <div
          className="airport airport--destination"
        >
          <span
            className="bpk-text bpk-text--lg airport__time"
          >
            08:25
          </span>
          <span
            className="bpk-text bpk-text--lg airport__name"
          >
            EDI
          </span>
        </div>
      </div>
      <div
        className="details"
      >
        <span
          className="bpk-text bpk-text--base details__duration"
        >
          1h 20
        </span>
        <span
          className="bpk-text bpk-text--base details__stops details__stops--single"
        >
          Direct
        </span>
      </div>
    </div>
    <div
      className="footer"
    >
      <div
        className="footer__info"
      >
        <span
          className="bpk-text bpk-text--xl footer__price"
        >
          321
           Â£
        </span>
        <span
          className="bpk-text bpk-text--base footer__agent"
        >
          ebookers
        </span>
      </div>
      <button
        className="bpk-button footer__button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Select
      </button>
    </div>
  </div>
</div>
`;
